packages/react-dom/src/events/DOMEventResponderSystem.js

```diff
@@ -110,7 +113,31 @@ const eventResponderContext: ResponderContext = {
       eventsWithStopPropagation.add(eventObject);
     }
   },
-  isPositionWithinTouchHitTarget(x: number, y: number): boolean {
+  isPositionWithinTouchHitTarget(doc: Document, x: number, y: number): boolean {
+    // This isn't available in some environments (JSDOM)
+    if (typeof doc.elementFromPoint !== 'function') {
+      return false;
+    }
+    const target = doc.elementFromPoint(x, y);
+    if (target === null) {
+      return false;
+    }
+    const childFiber = getClosestInstanceFromNode(target);
+    if (childFiber === null) {
+      return false;
+    }
+    const parentFiber = childFiber.return;
```

Comment:
Normally we avoid traversing Fibers outside the reconciler but we already cheat this rule in the existing event system - for the same reasons as here. So this is ok.